openmdao/core/group.py

```diff
@@ -1273,6 +1273,7 @@ def _solve_nonlinear(self):
         # TODO: It is pointless to run this ahead of non-iterative solvers.
         for sub in self.system_iter(recurse=True):
             if hasattr(sub, 'guess_nonlinear'):
+                self._transfer('nonlinear', 'fwd')
```

Comment:
Not sure how to find the correct isub, since we are iterating over a recursive list of subsystems. Is it stored somewhere keyed by pathname?